hermes_eea/calibration/calibration.py

```diff
@@ -184,26 +203,61 @@ def l0_sci_data_to_cdf(data: dict, original_filename: Path) -> Path:
     >>> data_packets = calib.parse_l0_sci_packets(data_filename)  # doctest: +SKIP
     >>> cdf_filename = calib.l0_sci_data_to_cdf(data_packets, data_filename)  # doctest: +SKIP
     """
+
+    # this is transferring name.bin to name.cdf
     file_metadata = parse_science_filename(original_filename.name)
 
+    # coarse = data["SHCOARSE"][idx]
+    # fine = data["SHFINE"][idx]
+    # time = convert_packet_time_to_datetime(coarse, fine)
     cdf_filename = original_filename.parent / create_science_filename(
         file_metadata["instrument"],
         file_metadata["time"],
         "l1",
         f'1.0.{file_metadata["version"]}',
     )
+    if not cdf_filename.is_file():
+        cdf = pycdf.CDF(
+            str(cdf_filename),
+            os.path.join(
+                hermes_eea._data_directory,
+                "masterSkeletons/hermes_eea_l1_00000000000000_v0.0.0.cdf",
+            ),
+        )
+        cdf.close()
+    if data:
+        cdf = pycdf.CDF(str(cdf_filename))
+        cdf.readonly(False)
 
-    # create an empty file for testing purposes
-    with open(cdf_filename, "w"):
-        pass
+        calibration_file = get_calibration_file(hermes_eea.stepper_table)
+        read_calibration_file(calibration_file)
 
-    return cdf_filename
+        #eea_cdf = WriteEEACDF(file_metadata, data_filename, hermes_eea.skeleton)
+        glblattr = Global_Attrs(file_metadata['version'],
+                                                  cdf_filename.name, lo_ext=False)
+        myEEA = EEA(file_metadata)
+        # This populates so doesn't have to return much
+        SkymapFactory(data, calib.energies, calib.deflections, myEEA)
+        most_active = np.where(np.array(myEEA.stats) > 150)
+        example_start_times = epoch_to_iso_obj(myEEA.Epoch[0:10])
+
+        n_packets = len(myEEA.Epoch)
+
+        hermes_eea_factory = Build_Hermes_EEA_Data(myEEA)
+        hermes_eea_factory.build_HermesData()
+
+        try:
+            cdf_path = hermes_eea_factory.hermes_eea_data.save( str(cdf_filename.parent) , True)
```

Comment:
This will likely fail when running on our pipeline as it is trying to save the file inside the python package, i think. The function should let the user set where the file will be saved.

Reply:
will do